core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -435,13 +437,11 @@ public void testConcurrentFastAppends(@TempDir File dir) throws Exception {
               for (int numCommittedFiles = 0;
                   numCommittedFiles < numberOfCommitedFilesPerThread;
                   numCommittedFiles++) {
-                while (barrier.get() < numCommittedFiles * threadsCount) {
-                  try {
-                    Thread.sleep(10);
-                  } catch (InterruptedException e) {
-                    throw new RuntimeException(e);
-                  }
-                }
+                final int currentFilesCount = numCommittedFiles;
+                Awaitility.await()
+                    .pollInterval(Duration.ofMillis(10))
+                    .pollInSameThread()
```

Comment:
why do we need this one here but not in the other places? Also this should have an `atMost()` configuration

Reply:
Only this test was becoming flaky with `Awaitility `. I think adding `pollInSameThread ` for all the similar tests is safe. So I have added `pollInSameThread ` and `atMost ` with 5 minutes to all the other similar tests. 
wdyt?

Reply:
@nk1506 I think the question was more around what's the issue if `pollInSameThread` is not used?

Reply:
@amogh-jahagirdar , My intent was to use the same test thread here. Since we have multiple runnable tasks here. I don't think it was flaky because of  not using `pollInSameThread()` . 
Please share your thoughts about adding `pollInSameThread()` here is a good idea or not.

Reply:
then we need to figure out why the test was flaky. Was it occassionally failing? The purpose of using Awaitility is to make tests more robust and less flaky, so we need to figure out why this is flaky

Reply:
If timeout is less`(<10 seconds)` it's failing randomly.  To safe side, I have added  timeout as 5 minutes `.atMost(Duration.ofSeconds(300))` .

Reply:
I'm -1 on having this run 5 minutes. Also I don't think we should be using `pollInSameThread()`, so please remove all of those. I ran those tests locally with the 10 seconds timeout consistently and they were passing